SpatialGDK/Documentation/content/helper-scripts.md

```diff
@@ -6,3 +6,4 @@ These scripts are located under `Plugins\UnrealGDK\SpatialGDK\Build\Scripts\` di
 | Helper script | Parameters | Description |
 | --- | --- | --- |
 | `BuildWorker.bat` | `<target> <platform> <configuration> <YourGame>.uproject [--skip-codegen]` | Example: </br> `BuildWorker.bat ExampleGameEditor Win64 Development ExampleGame.uproject` </br></br> Build, cook and zip your Unreal server-workers and client-workers for use with a SpatialOS cloud deployment (uploaded using [`spatial cloud upload`](https://docs.improbable.io/reference/latest/shared/deploy/deploy-cloud)). </br></br>  The following `<target>`s  generate zipped workers: </br> - `<YourGame>` </br> - `<YourGame>Server` </br> - `<YourGame>Editor` </br></br> Any other `<target>` passes all arguments to `Engine\Build\BatchFiles\Build.bat` with no cooking or zipping performed.|
+| `ExternalSchemaCodegen.bat` | `<schema_directory> <output_code_directory>` | Example: </br> `ExternalSchemaCodegen.bat spatial\schema\my_external_schema Game\Source\ThirdPersonShooter\ExternalSchemaCodegen` </br></br> Run the code generator binary created by the Setup.bat script to generate SpatialOS schema component types and an interface for sending and registering callbacks for receiving SpatialOS component updates.|
```

Comment:
I am not clear on a few issues.
1. Does this mean that this new script runs the code generator which was created by the `Setup.bat` script? 

2. Have we got anywhere to link to so we know what `Setup.bat` is about? We only have it listed as a way to update your Unreal Engine. Does it do anything else?

3. Which script generates schema?  `ExternalSchemaCodegen.bat` or `Setup.bat`? It's not clear

4. Rather than defining schema, we can link to the glossary definition.

5. The script name suggests that it's for external schema generation - should we mention this? It seems relevant?

6. Is the schema directory something which is already there? Or something I name when I run this script? Where can I find it? 

7. Is the output code something which is already there? Or something I name when I run this script? Where can I find it. Does it mean a schema file? I am not sure.

8. Is "the code generator binary" this script? Or something else that this script calls? Do we need to know it calls something else?

Also:  I have just updated the description for the other script on this page, so formatting this one similarly, 

```suggestion
| `ExternalSchemaCodegen.bat` | `<schema_directory> <output_code_directory>` | Use this script for components and callbacks you have created outside of the Unreal Editor. It generates [schema]({{urlRoot}}/content/glossary#schema) for these components and an interface for these callbacks</br></br>**Example** </br> `ExternalSchemaCodegen.bat spatial\schema\my_external_schema Game\Source\ThirdPersonShooter\ExternalSchemaCodegen` </br></br> **Flags**<br/>* `<schema_directory>` <Add info about this: Is it the directory for output,  is this something that's already there, and where it should be, and what should it be called if you have to create it> </br> * `<schema_directory>` < Add info about this: Is it the file for output,  is this something that's already there, where should it be, and what should it be called if you have to create it>  |
```

Reply:
Just noted that this is the same as the discussion here: https://docs.google.com/document/d/1DLrLcDsqrxhLpw1L_GNBGwoQR6kRjpWKr4-reUwsRYU/edit